analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantToArrayCall.CSharp11.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    public class RedundantToArrayCall
+    {
+        public void Utf8StringLiterals(int propertyValue)
+        {
+            var c = "some string"u8.ToArray()[10];        // Noncompliant
+            c = "some string"u8.Slice(5, 4)[1];           // Compliant
+            foreach (var v in "some string"u8.ToArray())  // Noncompliant {{Remove this redundant 'ToArray' call.}}
+            {
+                // ...
+            }
+
+            var arr = "some string"u8.ToArray(); // Compliant
+        }
+
+        public void ReadOnlySpans(int propertyValue)
```

Comment:
You can add a parameter `ReadOnlySpan<byte> span`. And like that, you don't have to create a new instance of the span and you will not need the unsafe block and the pointer and the test case will be much simpler.